src/Versity.Users/Core/Application/RequestHandlers/Users/Commands/ChangeUserPassword/ChangeUserPasswordCommand.cs

```diff
@@ -3,4 +3,4 @@
 
 namespace Application.RequestHandlers.Users.Commands.ChangeUserPassword;
 
-public record ChangeUserPasswordCommand(string Id, string OldPassword, string NewPassword) : IRequest<IdentityResult>;
+public record ChangeUserPasswordCommand(string OldPassword, string NewPassword, string Id = "") : IRequest<IdentityResult>;
```

Comment:
And why do you need to pass empty string here?

Reply:
I have two [endpoints](https://github.com/jejikeh/versity/blob/544049f665d3264f052ee63c7199a0e4c97cd231/src/Versity.Users/External/Presentation/Controllers/UsersController.cs#L39), one gets the userID from the JWT token, the second from the url.
I made two so that the admin could change passwords remotely, :\

When an object with an Empty ID is created, I get it from the JWT [here](https://github.com/jejikeh/versity/pull/12/files/ee64f37bbbf3993e8f32e9ae3c286e34c826ab00#diff-c87bc9d31cf6ee833c66d1962475c7bbf6637811430cb40cfdc9c17448367dac)

But I doubt it's even necessary. I can leave one that gets ID from the URL, just need to check the roles

Reply:
So far, the user does not know his ID. Its ID is stored only in the JWT token.

Reply:
Of course user will not know his id) But client will

Reply:
I [removed the check for an empty string](https://github.com/jejikeh/versity/pull/17/files#diff-c87bc9d31cf6ee833c66d1962475c7bbf6637811430cb40cfdc9c17448367dacR25), but now I have to check the ID from the JWT token and the role, due to the fact that the admin can change the passwords of other accounts.